ports/stm32/omv_csi.c

```diff
@@ -102,37 +106,19 @@ void omv_csi_mdma_irq_handler(void) {
 static int stm_csi_config(omv_csi_t *csi, omv_csi_config_t config) {
     if (config == OMV_CSI_CONFIG_INIT) {
         #if USE_DMA
-        // DMA Stream configuration
-        csi->dma.Instance = DMA2_Stream1;
-        #if defined(STM32H7)
-        csi->dma.Init.Request = DMA_REQUEST_DCMI;
-        #else
-        csi->dma.Init.Channel = DMA_CHANNEL_1;
-        #endif
-        csi->dma.Init.Direction = DMA_PERIPH_TO_MEMORY;
-        csi->dma.Init.MemInc = DMA_MINC_ENABLE;
-        csi->dma.Init.PeriphInc = DMA_PINC_DISABLE;
-        csi->dma.Init.PeriphDataAlignment = DMA_PDATAALIGN_WORD;
-        csi->dma.Init.MemDataAlignment = DMA_MDATAALIGN_WORD;
-        csi->dma.Init.Mode = DMA_NORMAL;
-        csi->dma.Init.Priority = DMA_PRIORITY_HIGH;
-        csi->dma.Init.FIFOMode = DMA_FIFOMODE_ENABLE;
-        csi->dma.Init.FIFOThreshold = DMA_FIFO_THRESHOLD_FULL;
-        csi->dma.Init.MemBurst = DMA_MBURST_INC4;
-        csi->dma.Init.PeriphBurst = DMA_PBURST_SINGLE;
-    
-        // Initialize the DMA stream
-        HAL_DMA_DeInit(&csi->dma);
-        if (HAL_DMA_Init(&csi->dma) != HAL_OK) {
+        // Configure and initialize DMA.
+        if (stm_dma_init(&csi->dma, OMV_CSI_DMA_CHANNEL, OMV_CSI_DMA_REQUEST,
+                    DMA_PERIPH_TO_MEMORY, true, OMV_CSI_DMA_XFER_PORTS, &stm_dma_csi_init)) {
```

Comment:
It looks like circular mode is getting set to true for the F4/F7/H7 versus normal mode.

Mode will end up being DMA_CIRCULAR versus DMA_NORMAL for the older boards.

Reply:
Actually, with `circular=False`, the DMA will be initialized immediately for the N6 it needs further initialization for circular mode. I tested `mode = DMA_CIRCULAR` and it seems to work fine for H7, or am I missing something?

Reply:
I think the circular arg should be true for the n6 and false for older boards. However, if circular works fine with the H7 plus I can test ont the F7 and F4. Was just pointing out the mode change.

Reply:
It does work, I think maybe it gets changed later in the driver. The problem that `circular` arg is how I can tell if N6 DMA is done or will need further initialization, without adding yet another arg that gets a value based on ifdef's.

Reply:
Yeah, tested F4 and F7 and they work fine. So, I guess it's fine to leave as is.